specification/security/resource-manager/Microsoft.Security/preview/2022-04-01-preview/query.json

```diff
@@ -0,0 +1,231 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Security Center",
+    "description": "API spec for Microsoft.Security (Azure Security Center) resource provider",
+    "version": "2022-04-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.Security/query": {
+      "post": {
+        "tags": [
+          "Query"
+        ],
+        "description": "Run KQL query against Microsoft.Security data types for scopes specified in the request. Supported types: Nodes, edges, insights, recommendations, vulnerabilities.",
+        "operationId": "Query",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "name": "query",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/QueryRequest"
+            },
+            "description": "Request specifying query, scope and its options."
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
>200 [](http://example.com/codeflow?start=11&length=3)

queries will never be async?  You will be able to meet the 2 second SLA for all queries?

Reply:
1 second at P99:  https://armwiki.azurewebsites.net/fundamentals/control_plane_kpis/control_plane_latency.html

Reply:
Our scenario/feature is different than "regular" APIs, it is not a core "Azure provider" one but kind of a "product" built on it. Also the latency we'll eventually land on is one that customers will be satisfied with. So I think we are good here.

Reply:
Appreciate the update on this. We will align to the required SLA.